core/templates/components/graph-services/graph-layout.service.spec.ts

```diff
@@ -0,0 +1,1207 @@
+        reachableFromEnd: true
+      },
+      State4: {
+        depth: 4,
+        offset: 0,
+        reachable: true,
+        x0: 0.07250000000000001,
+        y0: 0.66,
+        xLabel: 0.1625,
+        yLabel: 0.7,
+        id: 'State4',
+        label: 'State4',
+        height: 0.08,
+        width: 0.18000000000000002,
+        reachableFromEnd: true
+      },
+      State5: {
+        depth: 1,
+        offset: 3.5,
+        reachable: true,
+        x0: 0.8600000000000001,
+        y0: 0.26,
+        xLabel: 0.9500000000000001,
+        yLabel: 0.30000000000000004,
+        id: 'State5',
+        label: 'State5',
+        height: 0.08,
+        width: 0.18000000000000002,
+        reachableFromEnd: false
+      },
+      State6: {
+        depth: 2,
+        offset: 1.5,
+        reachable: true,
+        x0: 0.41000000000000003,
+        y0: 0.3933333333333333,
+        xLabel: 0.5,
+        yLabel: 0.43333333333333335,
+        id: 'State6',
+        label: 'State6',
+        height: 0.08,
+        width: 0.18000000000000002,
+        reachableFromEnd: false
+      },
+      State7: {
+        depth: 2,
+        offset: 2.5,
+        reachable: true,
+        x0: 0.6350000000000001,
+        y0: 0.3933333333333333,
+        xLabel: 0.7250000000000001,
+        yLabel: 0.43333333333333335,
+        id: 'State7',
+        label: 'State7',
+        height: 0.08,
+        width: 0.18000000000000002,
+        reachableFromEnd: false
+      },
+      State8: {
+        depth: 3,
+        offset: 0,
+        reachable: true,
+        x0: 0.07250000000000001,
+        y0: 0.5266666666666667,
+        xLabel: 0.1625,
+        yLabel: 0.5666666666666667,
+        id: 'State8',
+        label: 'State8',
+        height: 0.08,
+        width: 0.18000000000000002,
+        reachableFromEnd: true
+      },
+      State9: {
+        depth: 1,
+        offset: 0.5,
+        reachable: true,
+        x0: 0.185,
+        y0: 0.26,
+        xLabel: 0.275,
+        yLabel: 0.30000000000000004,
+        id: 'State9',
+        label: 'State9',
+        height: 0.08,
+        width: 0.18000000000000002,
+        reachableFromEnd: true
+      },
+      Orphaned: {
+        depth: 5,
+        offset: 0,
+        reachable: false,
+        x0: 0.07250000000000001,
+        y0: 0.7933333333333333,
+        xLabel: 0.1625,
+        yLabel: 0.8333333333333333,
+        id: 'Orphaned',
+        label: 'Orphaned',
+        height: 0.08,
+        width: 0.18000000000000002,
+        reachableFromEnd: false
+      }
+    };
+    let nodeData1 = cloneDeep(nodeData);
+
+    let graphWidth = sgls.getGraphWidth(
+      AppConstants.MAX_NODES_PER_ROW, AppConstants.MAX_NODE_LABEL_LENGTH);
+    let graphHeight = sgls.getGraphHeight(nodeData1);
+
+    let modifiedNodeData = sgls.modifyPositionValues(
+      nodeData, graphWidth, graphHeight);
+
+    expect(modifiedNodeData).not.toEqual(nodeData1);
+    expect(modifiedNodeData).toEqual(nodeData);
```

Comment:
Try not to do such a big comparison. I can't figure out what exactly we care about validating here.

Instead, validate the specific thing that you care about testing.

Reply:
Added specific checks for position values